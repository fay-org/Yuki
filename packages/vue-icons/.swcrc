{
  "jsc": {
    "experimental": {
      "plugins": [["swc-plugin-vue-jsx", {}]]
    }
  }
}
