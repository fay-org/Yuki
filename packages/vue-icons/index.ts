import { build } from './scripts/bundler'

//  iife ~
;(() => build())()
