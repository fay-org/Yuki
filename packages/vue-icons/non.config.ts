import path from 'path'
import { defineNonConfig } from '@fect-ui/cli'

export default defineNonConfig({
  lib: {
    input: path.join(__dirname, 'packages'),
    format: ['cjs', 'es']
  }
})
