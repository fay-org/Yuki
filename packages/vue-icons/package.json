{
  "name": "@fect-ui/vue-icons",
  "version": "2.0.1",
  "author": "XeryYue",
  "description": "Well prepared high quality Icon Library for fect",
  "scripts": {
    "build": "rm -rf dist types && ts-node scripts/bundler.ts"
  },
  "main": "dist/cjs/index.cjs.js",
  "module": "dist/esm/index.esm.js",
  "types": "types/index.d.ts",
  "license": "MIT",
  "files": [
    "LISCENSE",
    "dist",
    "types/",
    "README.md"
  ],
  "devDependencies": {
    "jsdom": "^16.6.0",
    "@types/svgo": "1.3.2",
    "svgo": "1.3.2",
    "@types/jsdom": "^16.2.11",
    "ora": "5.4.1"
  },
  "peerDependencies": {
    "vue": "^3.0.11"
  },
  "sideEffects": false,
  "homepage": "https://github.com/fect-org/fect",
  "repository": {
    "type": "git",
    "url": "https://github.com/fect-org/fect"
  }
}
