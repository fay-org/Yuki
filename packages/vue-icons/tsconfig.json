{
  "compilerOptions": {
    "baseUrl": ".",
    "outDir": "./lib",
    "target": "ES2017",
    "module": "CommonJS",
    "sourceMap": false,
    "declaration": true,
    "skipLibCheck": true,
    "esModuleInterop": true,
    "importHelpers": true,
    "allowJs": true,
    "resolveJsonModule": true,
    "jsx": "preserve",
    "lib": ["esnext", "dom"]
  },
  "include": ["scripts/**/*"]
}
