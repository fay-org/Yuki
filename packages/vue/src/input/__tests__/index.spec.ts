import { mount } from '@vue/test-utils'
import Input from '..'
import { Github } from '@fect-ui/vue-icons'

describe('Input', () => {
  it('should be render as a element', () => {
    const wrapper = mount(Input, {
      props: {
        modelValue: ''
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })
  it('normal props should work correctly', async () => {
    const wrapper = mount({
      components: {
        [Input.name]: Input
      },
      data() {
        return {
          suffix: '',
          prefix: '',
          disabled: false,
          readonly: false,
          clearable: false,
          type: 'password'
        }
      },
      template: `
          <div>
            <fe-input size="mini" :suffix="suffix" :prefix="prefix"
             :disabled="disabled" :clearable="clearable" 
             :type="type"
             placeholder="This is a test message">
              <span>Test</span>
            </fe-input>
          </div>
        `
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('label').html()).toContain('<label><span>Test</span></label>')
    await wrapper.setData({ disabled: true })
    expect(wrapper.find('.disabled').exists()).toBe(true)
  })

  it('when input focus should change border color', async () => {
    const wrapper = mount(Input)
    const el = wrapper.find('input')
    await el.trigger('focus')
    expect(wrapper.find('.fect-input__wrapper--hover').exists()).toBe(true)
    await el.trigger('blur')
    expect(wrapper.find('.fect-input__wrapper--hover').exists()).toBe(false)
  })

  it('when type as passowrd when trigger icon should be text', async () => {
    const wrapper = mount(Input, {
      props: {
        modelValue: '123456',
        type: 'password'
      }
    })

    const el = wrapper.find('.fect-input__icon')
    await el.trigger('click')
    expect(wrapper.find('input').attributes('type')).toBe('text')
    await el.trigger('click')
    expect(wrapper.find('input').attributes('type')).toBe('password')
  })

  it('should be support prefix and suffix label', () => {
    const wrapper = mount({
      template: `
      <div>
        <fe-input suffix="Suffix" prefix="Prefix"  />
      </div>
      `,
      components: {
        [Input.name]: Input
      }
    })
    expect(wrapper.find('.fect-input__label--prefix').exists()).toBe(true)
    expect(wrapper.find('.fect-input__label--suffix').exists()).toBe(true)
  })

  it('should support render prefix and suffix icon', () => {
    const wrapper = mount({
      components: {
        [Input.name]: Input,
        Github
      },
      template: `
        <div>
          <fe-input>
            <template #prefix-icon><Github/></template>
            <template #suffix-icon><Github/></template>
          </fe-input>
        </div>
      `
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('clearHandler should be work', async () => {
    const wrapper = mount(Input, {
      props: {
        modelValue: 333,
        clearable: true
      }
    })
    const el = wrapper.find('.fect-input__icon--suffix')
    await el.trigger('click')
    const emited = wrapper.emitted('update:modelValue')![0] as any
    expect(emited[0]).toBe('')
    await wrapper.setProps({ modelValue: '' })
    expect(wrapper.find('svg').exists()).toBe(false)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('changeHandler should be work', async () => {
    const wrapper = mount(Input, {
      props: {
        modelValue: 333,
        type: 'number'
      }
    })
    const input = wrapper.find('input')
    input.element.value = '2'
    await input.trigger('input')
    await input.trigger('change')
    expect(wrapper.emitted('update:modelValue')![0]).toEqual(['2'])
    expect(wrapper.emitted()).toHaveProperty('change')
  })
})
