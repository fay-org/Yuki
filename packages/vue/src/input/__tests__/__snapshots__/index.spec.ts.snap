// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input clearHandler should be work 1`] = `
<div class="fect-input fect-input--medium">
  <!---->
  <div class="fect-input__container">
    <!---->
    <div class="fect-input__wrapper">
      <!----><input type="text" class="">
      <div class="fect-input__icon fect-input__icon--suffix" style="cursor: default; pointer-events: none;">
        <!---->
      </div>
      <!---->
    </div>
    <!---->
  </div>
</div>
`;

exports[`Input normal props should work correctly 1`] = `
<div>
  <div class="fect-input fect-input--mini"><label><span>Test</span></label>
    <div class="fect-input__container">
      <!---->
      <div class="fect-input__wrapper">
        <!----><input type="password" class="" placeholder="This is a test message">
        <!---->
        <div class="fect-input__icon fect-input__icon--suffix" style="cursor: pointer; pointer-events: auto;"><svg viewBox="0 0 24 24" width="16" height="16" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision" style="color: currentColor;">
            <path d="M1 12s4-8 11-8 11 8 11 8-4 8-11 8-11-8-11-8z"></path>
            <circle cx="12" cy="12" r="3"></circle>
          </svg></div>
      </div>
      <!---->
    </div>
  </div>
</div>
`;

exports[`Input should be render as a element 1`] = `
<div class="fect-input fect-input--medium">
  <!---->
  <div class="fect-input__container">
    <!---->
    <div class="fect-input__wrapper">
      <!----><input type="text" class="">
      <!---->
      <!---->
    </div>
    <!---->
  </div>
</div>
`;

exports[`Input should support render prefix and suffix icon 1`] = `
<div>
  <div class="fect-input fect-input--medium">
    <!---->
    <div class="fect-input__container">
      <!---->
      <div class="fect-input__wrapper">
        <div class="fect-input__icon fect-input__icon--prefix" style="cursor: pointer; pointer-events: auto;"><svg fill="none" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.5" shape-rendering="geometricPrecision" viewBox="0 0 24 24" height="24" width="24" style="color: currentColor;">
            <path d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 00-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0020 4.77 5.07 5.07 0 0019.91 1S18.73.65 16 2.48a13.38 13.38 0 00-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 005 4.77a5.44 5.44 0 00-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 009 18.13V22"></path>
          </svg></div><input type="text" class="">
        <!---->
        <div class="fect-input__icon fect-input__icon--suffix" style="cursor: pointer; pointer-events: auto;"><svg fill="none" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.5" shape-rendering="geometricPrecision" viewBox="0 0 24 24" height="24" width="24" style="color: currentColor;">
            <path d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 00-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0020 4.77 5.07 5.07 0 0019.91 1S18.73.65 16 2.48a13.38 13.38 0 00-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 005 4.77a5.44 5.44 0 00-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 009 18.13V22"></path>
          </svg></div>
      </div>
      <!---->
    </div>
  </div>
</div>
`;
